react-router/lib/IndexRoute.d.ts

```diff
@@ -15,8 +15,8 @@ type ComponentsCallback = (err: any, components: RouteComponents) => any;
 export interface IndexRouteProps {
     component?: RouteComponent;
     components?: RouteComponents;
-    getComponent?(nextState: RouterState, callback: ComponentCallback): undefined;
```

Comment:
It is and it's not :)
It is enough from TS compiler's point of view, but the TSLint rule is broken at the moment and this makes CI to not pass. And leaving the `: undefined` variant now is not a solution, because it broke existing code and needs explicit `return undefined;` for all methods.

Optional properties with question mark do not really change anything functionally in this context (correct me if I'm wrong), because when implementing interface, no matter if you're declaring it as a method or as a property, you can use both notations for implementation. You can confirm it in a [TypeScript playground](https://www.typescriptlang.org/play/) with this code:
```ts
// interface IndexRouteProps {
//     getComponents?: (nextState: Object, callback: Function) => void;
// }

interface IndexRouteProps {
    getComponents?(nextState: Object, callback: Function): void;
}

class MyClass implements IndexRouteProps {
    getComponents(nextState, callback) {

    }
}

let impl: IndexRouteProps = {
    getComponents(nextState, callback) {

    }
};
```

Try switching between both interface declarations and you will see they both support both implementations.

And what it solves for us is the TSLint rule emitting the aforementioned error (see https://github.com/DefinitelyTyped/DefinitelyTyped/pull/14587#issuecomment-281160356).